var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -182,7 +184,11 @@ class Cmake(Package):
         depends_on('libuv@1.10.0:', when='@3.12.0:')
         depends_on('rhash', when='@3.8.0:')
 
-    with when('+ownlibs'):
+    with when('+ownlibs platform=darwin'):
```

Comment:
Apologies but this should also be done for `platform=cray`. This could be done in a loop like:

```
for x in ['cray', 'linux', 'darwin']:
  with when("+ownlibs platform={0}".format(x)):
    ...
```

It occurs to me we may want to add a shorthand for this (i.e. we can add an "on_all_platforms_except()" function/context-manager) but that doesn't have to be done in this PR.

Reply:
Several Spack packages currently use for-loops to instantiate groups of resources (for example) so using a for loop wouldn't be too out-of-place.